source/MaterialXGenShader/GenContext.cpp

```diff
@@ -26,6 +26,10 @@ GenContext::GenContext(ShaderGeneratorPtr sg) :
 
 void GenContext::addIdentifier(const string& name)
 {
+    if (name.empty())
```

Comment:
OSL had an empty string as a restricted name which doesn't really make sense. Instead leave the onus on identifier addition to check for invalid empty string. Tokens which also get added here should also never add an empty string (as is the case now with v-flip being substituted with an empty string.